pallets/merkle/Cargo.toml

```diff
@@ -36,7 +36,7 @@ default-features = false
 features = ["u64_backend", "alloc"]
 
 [dependencies.curve25519-gadgets]
-rev = "4028605"
+branch = "main"
```

Comment:
If there are breaking changes on `curve25519-gadgets`, it might not compile. Do you want it this way?

Reply:
I'll peg to a commit, good idea.